### DISCUSSION TITLE
Making select and multiselect fields user modifiable

### DISCUSSION THREAD

**Discussion by gzqn** [Budibase/budibase#4418] [Help]
I would like to to give users a way to update a select or multselect list. I have a project type field in  table which is just a categorisation, I need to filter on this categorisation so I don't want too many options, however as not all the necessary cateorisations are foreseeable I would like to give the option to select users to create a new category without giving them admin access. is this possible?

**Comment by mjashanks** [comment]
Hey @gzqn 

This is possible at the moment. Here's how you can do it...

1. Create a table that will store your options / lookups. I'll call this table `Lookups`. Your `Lookups` table could have columns
- description - a text value for the options
- value - an identifier for the option. You don't necessarily need this, but could allow you to change the description the lookup, without having to update lots of previously stored values 
- type - this will allow you to use this table for a range of different pickers. i.e. Type could be "COUNTRY" or "CITY"
2. In your design section, add a "Data Provider" around your Select. Set its "Data" to `Lookups`. At this point, you might want to also filter the data provider by the "Type" column. I.e. "Type = COUNTRY" if you only want to load COUNTRY options
3. Set your Select's "Options Source" to "Data Provider", and select your new data provider. Then set your value and label columns.
 
![image](https://user-images.githubusercontent.com/3524181/153578239-e98bff88-0d18-4fd5-a0a9-6833626df3cb.png)

Then, elsewhere, you could give your users access to a "Lookups" screen, where they can modify values.
